### DISCUSSION TITLE
What is the Unit of max_distance in sjoin_nearest

### DISCUSSION THREAD

**Discussion by fbirori** [geopandas/geopandas#2797] [Q&A] [answered]
I have a use case where I want to do a spatial join and get the nearest items that are within, for example,  `max_distance = 5 km `. The documentation for the ` max_distance ` parameter simply says ` Maximum distance within which to query for nearest geometry.` and it's a `float`.

What is the meaning of the value of ` max_distance ` from a distance/length on the globe (or plane) perspective? Is it the Euclidean distance between the geometries? How can I retrieve geometries that are within `max_distance = 5 km `

Thank you in advance for chipping in.

**Comment by brendan-ward** [comment, marked as answer]
Distance is in the same units as your coordinates.  If you are using a planar spatial reference system with coordinates in meters, you could set `max_distance=5000` to get 5km.

However, if your coordinates are in latitude / longitude, coordinates are treated as if they are planar rather than on a spheroid, and distances / areas are not meaningful (they do not account for spherical geometry, e.g., Great Circle distances).

You can either reproject your coordinates to a planar spatial reference system appropriate for your area of interest / analysis and use the existing methods in GeoPandas (which uses Shapely / GEOS under the hood, which are limited to planar coordinates), or use a different library.  The [spherely](https://github.com/benbovy/spherely) library based on Google's S2 may provide some of this capability in the future (spatial joins yes, not sure about nearest neighbor searches).

**Comment by fbirori** [comment]
Thank you @brendan-ward.  That answers my question. I pretty much had to do something like `df.to_crs(epsg = 3857, inplace = True)` which reprojected to a [meter-based coordinate system](https://epsg.io/3857).

I also found this question: [what-is-the-units-of-distance-col-and-max-distance-in-geopandas-sjoin-nearest](https://stackoverflow.com/questions/72569151/what-is-the-units-of-distance-col-and-max-distance-in-geopandas-sjoin-nearest) on StackOverflow to be helpful.
